### PR TITLE
Catch error if multiprocessing is not available or is unable to determine the number of CPUs

### LINKED ISSUE #4113: Installation fails on platforms without multiprocessing
On the latest version of requests in pypi, requests requires the multiprocessing module to install correctly, even though it is not used in the installation, only in the unit tests. This means it is no longer possible to install requests on platforms that lack multiprocessing support (such as Android or IoT devices)

## Expected Result

Requests is installed and available for use

## Actual Result

Installation fails with the following:

```
Downloading/unpacking requests
  Downloading requests-2.17.3.tar.gz (123Kb): 123Kb downloaded
  Running setup.py egg_info for package requests
    Traceback (most recent call last):
      File "<string>", line 14, in <module>
      File "/tmp/build/requests/setup.py", line 11, in <module>
        from multiprocessing import cpu_count
    ImportError: No module named multiprocessing
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):

  File "<string>", line 14, in <module>

  File "/tmp/build/requests/setup.py", line 11, in <module>

    from multiprocessing import cpu_count

ImportError: No module named multiprocessing

----------------------------------------
Command python setup.py egg_info failed with error code 1
Storing complete log in /sdcard/com.hipipal.qpyplus/.pip/pip.log
```

## Reproduction Steps

From a platform without multiprocessing:

```
pip install requests
```

## System Information

For obvious reasons, requests.help is not available, but I've tried to piece together the information that it would produce:

- platform.system()   Linux
- platform.release()   3.14.29
- implementation     CPython
- implementation_version   2.7.2
- ssl.OPENSSL_VERSION_NUMBER   268443727L
- requests_version   2.17.3

### PR DESCRIPTION
**Pull request by cyberjacob**
resolves #4113

### CODE REVIEW

**Comment by Lukasa** [comment]
FYI @kennethreitz, it seems only one check is running here: I have no idea why.

**Comment by sigmavirus24** [comment]
add to whitelist

**Comment by sigmavirus24** [comment]
(Taking a shot in the dark)

File: HISTORY.rst [hunk 1]
```diff
@@ -8,6 +8,10 @@ dev
 
 **Bugfixes**
 
+- Resolve instalation failure if multiprocessing is not available
```
Comment (Lukasa) [inline HISTORY.rst]: This should be `installation`.

**Comment by girlpunk** [comment]
There we go, spelling was never my forte (but the comment box here has spell check :)

**Comment by girlpunk** [comment]
no rush from me, I'm stuck with 2.15.1 until chardet/chardet#127 is sorted.

**Comment by kennethreitz** [comment]
@sigmavirus24 was a configuration issue, i fixed the problem. Sorry for the issues!

**Pull request merged**